hathor/merged_mining/coordinator.py

```diff
@@ -741,15 +754,10 @@ def job_request(self) -> None:
             self.send_request('mining.notify', job.to_stratum_params())
 
             # for debugging only:
-            bitcoin_block = job.build_bitcoin_block_header(self.dummy_work(job))
+            bitcoin_block = job.build_bitcoin_block_header(job.dummy_work())
```

Comment:
I know this line is not from this PR but should we do this calculus always? Even if it's just for debug?

This end up calling `build_merkle_root_from_path`, could this be a costly method? Maybe with the `dummy_work` is really fast but if it's not we could call this line only if we really are in debug

Reply:
`build_merkle_root_from_path` makes `N=log(len(transactions))` (typically  hashes of `N` txid's, those txid's are already in memory, so it's just calling `sha256d` `N` times in a row. Which is really really cheap. This also already being done (i.e. not new), and from the monitoring of the server it's pretty noticeable that resource usage is impressively low. So I'm worried with this right now, even if it's only for debugging.